glib/src/subclass/signal.rs

```diff
@@ -225,6 +221,92 @@ impl PartialEq for SignalId {
     }
 }
 
+#[derive(Copy, Clone, Hash)]
+pub struct SignalType(ffi::GType);
+
+impl SignalType {
+    pub fn with_static_scope(type_: Type) -> Self {
+        Self(type_.to_glib() | gobject_ffi::G_TYPE_FLAG_RESERVED_ID_BIT)
+    }
+
+    pub fn static_scope(&self) -> bool {
+        (self.0 & gobject_ffi::G_TYPE_FLAG_RESERVED_ID_BIT) != 0
+    }
+}
+
+impl From<Type> for SignalType {
+    fn from(type_: Type) -> Self {
+        Self(type_.to_glib())
+    }
```

Comment:
I'm not super sure about this, as the extra-bit for static scope is removed at `Type::from_glib` which means the `SignalType` might contain less information here. I wonder if it would make more sense to remove it only when doing `From<SignalType> for Type`

Reply:
Yes that could make sense, the you only need to be very careful to consistently use `SignalType` for all the signal related types.